bassa.py

```diff
@@ -0,0 +1,535 @@
+                raise ResponseError(
+                    'API response: {}'.format(result.status_code))
+            elif int(result.status_code/100) == 2:
+                retries = 0
+            elif int(result.status_code/100) == 5:
+                continue
+
+    def update_user_request(self, user_name=None, new_user_name=None, password=None, auth_level=None, email=None, retries=1):
+        """Update a user request
+        Args:
+            user_name (str):    Name of the user to updated
+            new_user_name (str): New name for the user
+            password (str): New password for the user
+            auth_level (int): Auth level for the new user, 0 for admins and 1 for regular users
+            email (str): Email ID for the new user
+            retires (int): number of retries upon failure
+        Returns:
+            None
+        """
+        if user_name is None or new_user_name is None or password is None or auth_level is None or email is None:
+            raise IncompleteParams
+        params = {}
+        params['user_name'] = new_user_name
+        params['password'] = password
+        params['auth_level'] = auth_level
+        params['email'] = email
+        endpoint = "/api/user"
+        api_url_complete = self.api_url + endpoint + "/" + user_name
+        while retries != -1:
+            result = requests.put(
+                api_url_complete, data=params, headers=self.headers)
+            retries = retries-1
+            if int(result.status_code/100) in [3, 4]:
+                retries = 0
+                raise ResponseError(
+                    'API response: {}'.format(result.status_code))
+            elif int(result.status_code/100) == 2:
+                retries = 0
+            elif int(result.status_code/100) == 5:
+                continue
+
+    def get_user_request(self, retries=1):
+        """Get a user request
+            retires (int): number of retries upon failure
+        Returns:
+            response as json
+        """
+        endpoint = "/api/user"
+        api_url_complete = self.api_url + endpoint
+        while retries != -1:
+            result = requests.get(api_url_complete, headers=self.headers)
+            retries = retries-1
+            if int(result.status_code/100) in [3, 4]:
+                retries = 0
+                raise ResponseError(
+                    'API response: {}'.format(result.status_code))
+            elif int(result.status_code/100) == 2:
+                retries = 0
+                return result.json()
+            elif int(result.status_code/100) == 5:
+                continue
+
+    def get_user_signup_requests(self, retries=1):
+        """Get all user requests
+        Args:
+            retires (int): number of retries upon failure
+        Returns:
+            response as json
+        """
+        endpoint = "/api/user/requests"
+        api_url_complete = self.api_url + endpoint
+        while retries != -1:
```

Comment:
Can we use some function to handle retry mechanism? You can also try to use a decorator for this.  This seems to be duplicated?

Reply:
Sure I will use timeouts, retries, request adapters, sessions other advanced features 
Will update it soon